domene/src/main/kotlin/no/nav/tiltakspenger/domene/behandling/Søknadsbehandling.kt

```diff
@@ -84,79 +88,71 @@ sealed interface Søknadsbehandling : Behandling {
                 it.lagVurdering(vurderingsperiode)
             }
 
             // Etter at vi har laget vurderinger, sjekker vi utfallet
-
-            if (vurderinger.any { it.utfall == Utfall.KREVER_MANUELL_VURDERING }) {
-                return BehandlingVilkårsvurdert.Manuell(
-                    id = id,
-                    sakId = sakId,
-                    søknader = søknader,
-                    vurderingsperiode = vurderingsperiode,
-                    saksopplysninger = saksopplysninger,
-                    tiltak = tiltak,
-                    vilkårsvurderinger = vurderinger,
-                    saksbehandler = saksbehandler,
-                )
-            }
-            if (vurderinger.any { it.utfall == Utfall.KREVER_MANUELL_VURDERING }) {
-                return BehandlingVilkårsvurdert.Manuell(
-                    id = id,
-                    sakId = sakId,
-                    søknader = søknader,
-                    vurderingsperiode = vurderingsperiode,
-                    saksopplysninger = saksopplysninger,
-                    tiltak = tiltak,
-                    vilkårsvurderinger = vurderinger,
-                    saksbehandler = saksbehandler,
-                )
-            }
-            if (vurderinger.all { it.utfall == Utfall.OPPFYLT }) {
-                return BehandlingVilkårsvurdert.Innvilget(
-                    id = id,
-                    sakId = sakId,
-                    søknader = søknader,
-                    vurderingsperiode = vurderingsperiode,
-                    saksopplysninger = saksopplysninger,
-                    tiltak = tiltak,
-                    vilkårsvurderinger = vurderinger,
-                    saksbehandler = saksbehandler,
-                )
+            return when {
+                vurderinger.any { it.utfall == Utfall.KREVER_MANUELL_VURDERING } ->
+                    BehandlingVilkårsvurdert.Manuell(
+                        id = id,
+                        sakId = sakId,
+                        søknader = søknader,
+                        vurderingsperiode = vurderingsperiode,
+                        saksopplysninger = saksopplysninger,
+                        tiltak = tiltak,
+                        vilkårsvurderinger = vurderinger,
+                        saksbehandler = saksbehandler,
+                    )
+
+                vurderinger.all { it.utfall == Utfall.OPPFYLT } ->
+                    BehandlingVilkårsvurdert.Innvilget(
+                        id = id,
+                        sakId = sakId,
+                        søknader = søknader,
+                        vurderingsperiode = vurderingsperiode,
+                        saksopplysninger = saksopplysninger,
+                        tiltak = tiltak,
+                        vilkårsvurderinger = vurderinger,
+                        saksbehandler = saksbehandler,
+                    )
+
+                vurderinger.all { it.utfall == Utfall.IKKE_OPPFYLT } ->
```

Comment:
Ikke noe logikk som du har endret på her ser det ut som, men skulle denne ha brukt `any` istedetfor `all`? Holder vel at én av vurderingene er "ikke oppfylt" for at det skal bli avslag.

Reply:
Enig. Kanskje man ikke merker det pga else-en nedenfor?